pluginlib/include/pluginlib/class_loader_imp.hpp

```diff
@@ -516,7 +516,10 @@ std::string ClassLoader<T>::getClassLibraryPath(const std::string & lookup_name)
       return *it;
     }
   }
-  return "";
+  std::ostringstream error_msg;
+  error_msg << "Could not find library corresponding to plugin " << lookup_name <<
+    ". Make sure that the library '" << library_name << "' actually exists.";
+  throw pluginlib::LibraryLoadException(error_msg.str());
```

Comment:
This changes the behavior of public API, no? Shouldn't we avoid that in a released rosdistro?